plugins/scaffolder/src/components/JobStatusModal/JobStatusModal.tsx

```diff
@@ -43,10 +43,11 @@ export const JobStatusModal = ({
   entity,
 }: Props) => {
   const job = useJobPolling(jobId);
+  const jobStatus = job?.status ?? '';
 
   useEffect(() => {
-    if (job?.status === 'COMPLETED') onComplete(job);
-  }, [job, onComplete]);
+    if (jobStatus === 'COMPLETED') onComplete(job!);
+  }, [jobStatus, onComplete]); // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
hm so job isn't recreated on each return from useJobPolling? this missing dep does look like an actual risk since the effect function may be bound to an old version of the job